valhalla/midgard/gridded_data.h

```diff
@@ -395,15 +390,18 @@ template <std::size_t dimensions_t> class GriddedData : public Tiles<PointLL> {
           Polyline2<PointLL>::Generalize(line, gen_factor, {});
         }
         // if this ends up as an inner we'll undo this later
-        if (cache[&line] > 0) {
-          line.reverse();
-        }
+        //        if (cache[&line] > 0) {
+        //          line.reverse();
+        //        }
         // sampling the bottom left corner means everything is skewed, so unskew it
         for (auto& coord : line) {
           coord.first += h;
           coord.second += h;
         }
       }
+      // remove points and lines
+      contour.remove_if([](const contour_t& line) { return line.size() < 4; });
```

Comment:
This has nothing to with current PR, just sanity improvement.
When `generalize` != 0 some of the inner holes turn into lines and points. I think we should remove them 
![Screen Shot 2021-03-11 at 13 39 21](https://user-images.githubusercontent.com/14297128/110774899-5a209e80-826f-11eb-84c9-0d3c37a84f60.png)

Reply:
agree